gradio/processing_utils.py

```diff
@@ -259,21 +268,33 @@ def save_file_to_cache(file_path: str | Path, cache_dir: str) -> str:
 
     return full_temp_file_path
 
+def check_public_url(url: str):
+    hostname = urlparse(url).hostname
+    if not hostname:
+        raise httpx.RequestError(f"Invalid URL: {url}")
+    ip = socket.gethostbyname(hostname)
```

Comment:
I wasn't familiar with this function, so looked it up: https://docs.python.org/3/library/socket.html

> Translate a host name to IPv4 address format. The IPv4 address is returned as a string, such as '100.50.200.5'. If the host name is an IPv4 address itself it is returned unchanged. See [gethostbyname_ex()](https://docs.python.org/3/library/socket.html#socket.gethostbyname_ex) for a more complete interface. [gethostbyname()](https://docs.python.org/3/library/socket.html#socket.gethostbyname) does not support IPv6 name resolution, and [getaddrinfo()](https://docs.python.org/3/library/socket.html#socket.getaddrinfo) should be used instead for IPv4/v6 dual stack support.

So I think we'll need to switch to `getaddrinfo` instead